### ISSUE TITLE
 Can't resolve 'd3-format' in 

### ISSUE THREAD

**Issue by theunreal** [swimlane/ngx-charts#431]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here
```

**Current behavior**
Error when building the app using this module:

```
/~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-format' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:27-47
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-array' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:141-160
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-brush' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:162-181
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-color' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:183-202
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-force' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:204-223
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-hierarchy' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:257-280
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-interpolate' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:282-307
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-scale' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:309-328
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-selection' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:330-353
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-shape' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:355-374
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-time-format' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:376-401
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
2client?ffdb:41 [WDS] App updated. Recompiling...
client?ffdb:41 [WDS] Errors while compiling. Reload prevented.
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-format' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:27-47
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-array' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:141-160
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-brush' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:162-181
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-color' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:183-202
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-force' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:204-223
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-hierarchy' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:257-280
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-interpolate' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:282-307
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-scale' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:309-328
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-selection' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:330-353
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-shape' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:355-374
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-time-format' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:376-401
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
2client?ffdb:41 [WDS] App updated. Recompiling...
client?ffdb:41 [WDS] Errors while compiling. Reload prevented.
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-format' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:27-47
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-array' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:141-160
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-brush' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:162-181
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-color' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:183-202
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-force' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:204-223
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-hierarchy' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:257-280
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-interpolate' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:282-307
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-scale' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:309-328
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-selection' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:330-353
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-shape' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:355-374
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
errors @ client?ffdb:119
sock.onmessage @ socket.js:37
EventTarget.dispatchEvent @ eventtarget.js:51
(anonymous) @ main.js:274
SockJS._transportMessage @ main.js:272
EventEmitter.emit @ emitter.js:50
WebSocketTransport.ws.onmessage @ websocket.js:35
wrapFn @ zone.js:1056
ZoneDelegate.invokeTask @ zone.js:424
Zone.runTask @ zone.js:191
ZoneTask.invoke @ zone.js:486
client?ffdb:119 ./~/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-time-format' in 'c:\xampp\htdocs\iot2\node_modules\@swimlane\ngx-charts\release'
 @ ./~/@swimlane/ngx-charts/release/index.js 8:376-401
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:4200 ./src/main.ts
```

**Expected behavior**
Build success

**Reproduction of the problem**
Just installing and importing the module

**What is the motivation / use case for changing the behavior?**
Bug fix

**Please tell us about your environment:**
windows,npm, latest angular (4)

* **ngx-charts version:** 5.3.1

* **Angular version:** 4.1.2

* **Browser:** All

* **Language:** TypeScript

**Comment by dcasier** [comment]
npm install --save d3 ?

**Comment by marjan-georgiev** [comment]
Yes, d3 is a peer dependency of ngx-charts, so you will need to install it in your project. I'll close this. Please let know if this does not fix your problem.

**Comment by joaogarin** [comment]
I am getting the same error also, even with d3 installed. Also included @types/d3 but doesn't seem to help..

````
"d3": "^4.9.1",
````

````
"@types/d3": "4.9.0",
````

**Comment by joaogarin** [comment]
I reinstalled d3 and it worked..so might have been some caching issue.

**Comment by GallinEnerg** [comment]
Also needed to reinstall d3! as per  dcasier

**Comment by AsmaGargouri** [comment]
I have a similar problem. 

`./node_modules/@swimlane/ngx-charts/release/index.js
Module not found: Error: Can't resolve 'd3-array' in '...\node_modules\@swimlane\ngx-charts\release'
 @ ./node_modules/@swimlane/ngx-charts/release/index.js 8:151-170
 @ ./src/app/app.module.ts
 @ ./src/main.ts
 @ multi webpack-dev-server/client?http://localhost:3000 ./src/main.ts`

Do we have to install d3 ?

**Comment by marjan-georgiev** [comment]
@AsmaGargouri yes.

**Comment by bhuvanchawla** [comment]
i have same issue what should i do for this

**Comment by Marimuthu-Aathi** [comment]
> i have same issue what should i do for this

I reinstalled d3 and it worked for me. May be have been some caching issue

**Comment by Sammion** [comment]
Fixed with install d3@6.7.0 but not work with d3@6.3
